src/helpers/insight-matrix/value-calculator.ts

```diff
@@ -0,0 +1,23 @@
+export interface InsightMatrixValueCalculator {
+    calculate: (thresholdDays: string[], values: number[]) => number | undefined;
+}
+
+export const createAverageValueInsightMatrixValueCalculator = (): InsightMatrixValueCalculator => {
+    return {
+        calculate: (_: string[], values: number[]) => {
+            const nonZeroValues = values.filter(v => v > 0);
+            return nonZeroValues.length > 0 ? nonZeroValues.reduce((total, current) => total + current, 0) / nonZeroValues.length : undefined;
```

Comment:
Something about filtering out zero values feels off.  bc in normal circumstances, the count of all values makes up the denominator. Can this be renamed to include the fact that it ignores non zeros? thinking about consuming this downstream, it could create confusion.

Reply:
The reason the non-zero values are being removed here is because some device types like Fitbit report a value whether you wear your device or not.  Steps, for example, gets reported as zero for each day you didn't wear it.  It doesn't mean that you walked zero steps that day.  It just means it didn't record any.  If we include the zero values, it can greatly affect the averages and not accurately reflect how much you actually walk each day.

I think zero is probably not a valid value for the majority of device data types, since most devices don't report a value at all when it is zero, but I also don't have a problem with renaming this implementation to be very clear that it will filter out the zero values when computing the average.

Reply:
Done.  🙂